File: 15-02-2026/figure_skating.py
def compute_score(judge_scores, *penalties):

    return sum(sorted(judge_scores)[1:-1]) - sum(penalties) or 0

print(compute_score([10, 8, 9, 6, 9, 8, 8, 9, 7, 7], 1)); # 64
print(compute_score([10, 10, 10, 10, 10, 10, 10, 10, 10, 10])); # 80
print(compute_score([10, 8, 9, 10, 9, 8, 8, 9, 10, 7], 1, 2, 1)); # 67
print(
  compute_score([8.0, 8.5, 9.0, 8.5, 9.0, 8.0, 9.0, 8.5, 9.0, 8.5], 0.5, 1.0),
); # 67.5

File: 15-02-2026/figure_skating.js
const computeScore = (judgeScores, ...penalties) =>
  judgeScores
    .sort((a, b) => a - b)
    .slice(1, judgeScores.length - 1)
    .reduce((acc, cur) => acc + cur) -
  (penalties.length > 1
    ? penalties.reduce((acc, cur) => acc + cur)
    : penalties[0] || 0);

console.log(computeScore([10, 8, 9, 6, 9, 8, 8, 9, 7, 7], 1)); // 64
console.log(computeScore([10, 10, 10, 10, 10, 10, 10, 10, 10, 10])); // 80
console.log(computeScore([10, 8, 9, 10, 9, 8, 8, 9, 10, 7], 1, 2, 1)); // 67
console.log(
  computeScore([8.0, 8.5, 9.0, 8.5, 9.0, 8.0, 9.0, 8.5, 9.0, 8.5], 0.5, 1.0),
); // 67.5
